### PR TITLE
check for missingness

### LINKED ISSUE #8: handling of missing data
Currently, one cannot have missing values even in variables that are not used for matching.  This should be fixed so that the function only checks the existence of missing values for the variables that are used.

### PR DESCRIPTION
**Pull request by alexWhitworth**
we add checks for missingness to fix #8 including tests that our methods work. More robust testing would be appreciated if anyone has suggestions.

Also, *importantly* we increment the version number in the `DESCRIPTION` file. Version numbering has not been updated with recent bug fixes--causing downstream problems in debugging problems related to making sure server installed software matches local software versions.

### CODE REVIEW

**Comment by kosukeimai** [comment]
@alexWhitworth The problem is that we may have missing data in some other inputs such as `exact` . That's why we were hesitant to do missing data checking for this package.  We need to come up with a more comprehensive way to do checking.

**Comment by alexWhitworth** [comment]
@kosukeimai I think you misunderstand what this commit does. Or, perhaps I'm misunderstanding your comment, in which case can you please clarify?

Let's say we have `data.frame` `dt` with columns `Y` = {0,1}, `x_1, ..., x_n` which are continuous data without missingness and `z_1, ..., z_n` which are continuous variables with missingness. 

- Currently `matchit(Y ~ x_1 + ..., x_n, data= dt)` will fail with error "data has missing values" even though the data in the call to `matchit` has no missingness. This is a code error.
- `matchit(Y ~ x_1 + ... + x_n + z_1 + ... + z_n, data= dt)` will also fail. And it should-- we're trying to maching using data with missingness. This is a methodological issue.

My commit fixes the first case but doesn't touch the second.

**Comment by kosukeimai** [comment]
The problem is that `var_idx` only grabs variables from `formula` but some variables may be used in other parts of the input to `matchit()`

**Comment by alexWhitworth** [comment]
Ah, thanks! Can you elaborate or point to a some line(s) of code?

File: R/matchit.R [hunk 1]
```diff
@@ -1,20 +1,29 @@
...
+  if (!inherits(formula, "formula")) { formula <- formula(formula) }
+  if(is.null(data)) stop("Dataframe must be specified",call.=FALSE)
+  if(!is.data.frame(data)){
+    stop("Data must be a dataframe",call.=FALSE)}
+  
+  varnames <- all.vars(formula)
+  var_idx <- which(names(data) %in% varnames)
+  if (length(varnames) == 0) stop("formula does not contain variables")
+  if ( any(!(varnames %in% names(data))) ) stop("formula variables do not exist in data.")
+  if (any(apply(data[, var_idx], 2, function(j) sum(is.na(j) || is.nan(j))) > 0))
```
Comment (kosukeimai) [inline R/matchit.R]: This line only works if all variables are contained in `var_idx` but in some cases (like nearest neighbor matching with exact constraint) other inputs contain variables and so we need to check the existence of missing values in those variables too.

**Pull request closed without merging**